### PR TITLE
Fix vending machines showing premium as free to dept workers

### PR DESCRIPTION
**Pull request by JJRcop**
## About The Pull Request
Vending machines showed premium products as free if you worked in the same department the vending machine payed out to. They are supposed to only show normal items as free.

This was just an interface bug and the vending machines still took your money for the premium items.

## Changelog
:cl: JJRcop
fix: Fixed vending machines showing premium products as free to workers of their department.
/:cl:

### CODE REVIEW

**Review by stylemistake — DISMISSED** [review]

**Comment by actioninja** [comment]
I like this solution better then the one I made without realizing this was up. just add preimum to the contraband items as well.

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -690,7 +691,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 			price = R.custom_price || default_price,
 			max_amount = R.max_amount,
 			ref = REF(R),
-			extended = TRUE
+			premium = TRUE
```
Comment (Cyberboss) [inline code/modules/vending/_vending.dm]: This change needs explaining to me.

Comment (JJRcop) [reply code/modules/vending/_vending.dm]: This var wasn't used in the interface, and contraband items are also not exempt from department members as @actioninja mentioned, so their price needs to be shown too.

Comment (actioninja) [reply code/modules/vending/_vending.dm]: yeah, this is correct.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**